### DISCUSSION TITLE
Use of fortran_unit_testing

### DISCUSSION THREAD

**Discussion by rljacob** [ESMCI/cime#4378] [General]
Are the tests invoked by scripts/fortran_unit_testing being run on a regular basis by anyone?

I know E3SM isn't running them.

@billsacks might know.

**Comment by rljacob** [comment]
Specifically is CESM running the tests in https://github.com/ESCOMP/CESM_share/tree/main/test/unit on a regular basis?

**Comment by billsacks** [comment]
We do use scripts/fortran_unit_testing regularly / frequently in CTSM testing. The CESM_share unit tests are not regularly run, though I run them on an as-needed basis. I'd like for us to run them more regularly (see https://github.com/ESCOMP/CESM_share/issues/20).

I'm curious what your motivation is: were you thinking about whether this could be removed, or perhaps leveraged more heavily? I forget: do you have a different system for running Fortran unit tests in E3SM (by which I mean traditional software unit tests that generate a pass/fail result)?

**Reply by rljacob** [reply]
I'm fine with where it is.   We want to start running it and I was checking that this wasn't dead code.  We want to add test for the timing library in our copy of share.   We don't have any other system for running Fortran unit tests (there were some attempts that didn't pan out).

**Reply by billsacks** [reply]
Sounds great - I'm glad to hear that it will get more use (= more people to help make sure it keeps working).
